query_executor.go

```diff
@@ -50,17 +50,41 @@ func (q *queryExecutor) executeQuery(qry ExecutableQuery) (*Iter, error) {
 		}
 
 		iter = q.attemptQuery(qry, conn)
-
 		// Update host
 		hostResponse.Mark(iter.err)
 
+		if rt == nil {
+			break
+		}
+
+		switch rt.GetRetryType(iter.err) {
+		case Retry:
+			for rt.Attempt(qry) {
```

Comment:
I would rather refactor the outerloop so that it takes care of this and there isn't as much branching.